jdaviz/configs/mosviz/tests/test_data_loading.py

```diff
@@ -0,0 +1,225 @@
+# Tests data loading in the Mosviz Jdaviz configuration
+
+import pathlib
+from tempfile import gettempdir
+from zipfile import ZipFile
+
+from astropy.nddata import CCDData
+from astropy.utils.data import download_file
+import numpy as np
+import pytest
+from specutils import Spectrum1D
+
+
+def test_load_spectrum1d(mosviz_helper, spectrum1d):
+    label = "Test 1D Spectrum"
+    mosviz_helper.load_1d_spectra(spectrum1d, data_labels=label)
+
+    assert len(mosviz_helper.app.data_collection) == 2
+    assert mosviz_helper.app.data_collection[0].label == label
+
+    table = mosviz_helper.app.get_viewer('table-viewer')
+    table.widget_table.vue_on_row_clicked(0)
+
+    data = mosviz_helper.app.get_data_from_viewer('spectrum-viewer')
+
+    assert isinstance(list(data.values())[0], Spectrum1D)
+    assert list(data.keys())[0] == label
+
+    with pytest.raises(TypeError):
+        mosviz_helper.load_1d_spectra([1, 2, 3])
+
+
+@pytest.mark.filterwarnings('ignore')
+def test_load_image(mosviz_helper, mos_image):
+    label = "Test Image"
+    mosviz_helper.load_images(mos_image, data_labels=label)
+
+    assert len(mosviz_helper.app.data_collection) == 2
+    assert mosviz_helper.app.data_collection[0].label == f"{label} 0"
+
+    table = mosviz_helper.app.get_viewer('table-viewer')
+    table.widget_table.vue_on_row_clicked(0)
+
+    data = mosviz_helper.app.get_data_from_viewer('image-viewer')
+
+    assert isinstance(list(data.values())[0], CCDData)
+    assert list(data.values())[0].shape == (55, 55)
+    assert list(data.keys())[0] == f"{label} 0"
```

Comment:
Might also want to consider data shape that isn't square so you can also catch any accidental transpose.

```suggestion
    dataval = data[label]
    assert isinstance(dataval, CCDData)
    assert dataval.shape == (55, 55)
```